rest-api/dao/questionnaire_response_dao.py

```diff
@@ -200,8 +201,11 @@ def _update_participant_summary(
         if code:
           summary_field = QUESTION_CODE_TO_FIELD.get(code.value)
           if summary_field:
-            something_changed = self._update_field(participant_summary, summary_field[0],
-                                                   summary_field[1], answer)
+            if something_changed:
+              self._update_field(participant_summary, summary_field[0], summary_field[1], answer)
+            else:
+              something_changed = self._update_field(participant_summary, summary_field[0],
+                                                     summary_field[1], answer)
```

Comment:
The same function gets called whether something_changed is True or False ?

Reply:
Yes, the same function.  The something_changed flag should be set to TRUE if anything changed, if it's already TRUE, it means something changed, it should not set back to false anymore.